--- a/package/base-files/files/lib/upgrade/common.sh
+++ b/package/base-files/files/lib/upgrade/common.sh
@@ -237,6 +237,10 @@ do_upgrade() {
 		platform_copy_config
 	fi
 
+	if type 'platform_copy_licenses' >/dev/null 2>/dev/null; then
+		platform_copy_licenses
+	fi
+
 	v "Upgrade completed"
 	[ -n "$DELAY" ] && sleep "$DELAY"
 	ask_bool 1 "Reboot" && {
--- a/target/linux/x86/base-files/lib/upgrade/platform.sh
+++ b/target/linux/x86/base-files/lib/upgrade/platform.sh
@@ -55,6 +55,16 @@ platform_copy_config() {
 	fi
 }
 
+platform_copy_licenses() {
+	local rootfs="$(x86_get_rootfs)"
+	local rootfsdev="${rootfs##*:}"
+	
+	mount -t ext4 -o rw,noatime "${rootfsdev%[0-9]}1" /mnt
+	cp -af "/etc/licenses" /mnt/
+	cp -af "/etc/product_id.txt" /mnt/
+	umount /mnt
+}
+
 platform_do_upgrade() {
 	platform_export_bootpart
 
