Index: build_dir/target/linux/x86/base-files/lib/upgrade/platform.sh
===================================================================
--- build_dir.orig/target/linux/x86/base-files/lib/upgrade/platform.sh
+++ build_dir/target/linux/x86/base-files/lib/upgrade/platform.sh
@@ -44,6 +44,16 @@ platform_copy_config() {
 	fi
 }
 
+platform_copy_licenses() {
+	local rootfs="$(x86_get_rootfs)"
+	local rootfsdev="${rootfs##*:}"
+	
+	mount -t ext4 -o rw,noatime "${rootfsdev%[0-9]}1" /mnt
+	cp -af "/etc/licenses" /mnt/
+	cp -af "/etc/product_id.txt" /mnt/
+	umount /mnt
+}
+
 platform_do_upgrade() {
 	local diskdev partdev diff
 
