--- a/package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh
+++ b/package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh
@@ -458,7 +458,12 @@ mac80211_prepare_vif() {
 			}
 		;;
 		mesh)
-			iw phy "$phy" interface add "$ifname" type mp
+ 			json_get_vars mesh_id
+			# Originally mesh_id is set by meshd-nl80211 if authsae is used, but---
+			# Scott changed to always add mesh_id so the mesh params can be set later
+			# and avoid link has been severed errors. The kernel will not allow
+			# mesh params to be set with no mesh_id
+			iw phy "$phy" interface add "$ifname" type mp mesh_id "$mesh_id"
 		;;
 		monitor)
 			iw phy "$phy" interface add "$ifname" type monitor
@@ -592,7 +597,7 @@ mac80211_setup_vif() {
 	case "$mode" in
 		mesh)
 			# authsae or wpa_supplicant
-			json_get_vars key
+			json_get_vars key mesh_id
 			if [ -n "$key" ]; then
 				if [ -e "/lib/wifi/authsae.sh" ]; then
 					. /lib/wifi/authsae.sh
