--- a/package/network/services/authsae/files/lib/wifi/authsae.sh
+++ b/package/network/services/authsae/files/lib/wifi/authsae.sh
@@ -45,6 +45,15 @@ authsae:
 };
 EOF
 
+	# Ensure that meshd isn't already running - not sure why this happens sometimes -- Scott 07-01-2014!
+	if [ -f /var/run/authsae-$ifname.pid ] ; then
+		old_pid=`cat /var/run/authsae-$ifname.pid`
+		kill -0 $old_pid
+		if [ "$?" == "0" ] ; then
+			echo "Meshd process for $ifname, pid $old_pid was already running! Stopping it..."
+			kill $old_pid
+		fi
+	fi
 	/usr/bin/meshd-nl80211 -i "$ifname" -s "$mesh_id" -c "$authsae_conf_file" </dev/null >/dev/null 2>/dev/null &
 	authsae_pid="$!"
 	ret="$?"
@@ -53,5 +62,6 @@ EOF
 	wireless_add_process "$authsae_pid" "/usr/bin/meshd-nl80211" 1
 
 	[ "$ret" != 0 ] && wireless_setup_vif_failed AUTHSAE_FAILED
+	[ "$ret" != 0 ] && echo "Wireless setup frailed for $ifname, pid $authsae_pid"
 	return $ret
 }
