--- a/package/network/utils/iproute2/Makefile
+++ b/package/network/utils/iproute2/Makefile
@@ -28,7 +28,7 @@ define Package/iproute2/Default
   URL:=http://www.linuxfoundation.org/collaborate/workgroups/networking/iproute2
   SUBMENU:=Routing and Redirection
   MAINTAINER:=Russell Senior <russell@personaltelco.net>
-  DEPENDS:= +libnl-tiny
+  DEPENDS:= +libnl
 endef
 
 define Package/ip-tiny
@@ -80,7 +80,7 @@ $(call Package/iproute2/Default)
 endef
 
 ifeq ($(BUILD_VARIANT),tiny)
-  IP_CONFIG_TINY:=y
+  IP_CONFIG_TINY:=n
 endif
 
 define Build/Configure
@@ -96,7 +96,7 @@ endef
 TARGET_CFLAGS += -ffunction-sections -fdata-sections
 
 MAKE_FLAGS += \
-	EXTRA_CCOPTS="-I$(STAGING_DIR)/usr/include/libnl-tiny" \
+	EXTRA_CCOPTS="-I$(STAGING_DIR)/usr/include/libnl" \
 	KERNEL_INCLUDE="$(LINUX_DIR)/include" \
 	SHARED_LIBS="" \
 	LDFLAGS="$(TARGET_LDFLAGS) -Wl,--gc-sections" \
