--- a/package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh
+++ b/package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh
@@ -486,7 +486,7 @@ mac80211_prepare_vif() {
 	esac
 
 	# Ensure the interface is down
-	ip link show dev "$ifname" | grep ',UP' > /dev/null
+	ip link show dev "$ifname" 2>/dev/null | grep ',UP' > /dev/null
 	if [ "$?" = "0" ] ; then
 		echo "Interface $ifname is up prematurely! Taking it back down..."
 		ip link set "$ifname" down
@@ -608,7 +608,16 @@ mac80211_setup_vif() {
 	json_get_vars mode
 	json_get_var vif_txpower txpower
 
-	ip link set dev "$ifname" up || {
+	# hostapd needs some extra time if it's doing auto channel or DFS
+	retries=60
+	logger "Waiting up to 60 seconds for radio to acquire channel..."
+	while [ "$retries" -ge 0 ] ; do
+		ip link set dev "$ifname" up > /dev/null 2>&1 && break
+		sleep 1
+		let retries=retries-1
+	done
+
+	[ "$retries" -lt 0 ] && {
 		wireless_setup_vif_failed IFUP_ERROR
 		json_select ..
 		return
@@ -709,7 +718,7 @@ mac80211_interface_cleanup() {
 
 	for wdev in $(list_phy_interfaces "$phy"); do
 		ip link set dev "$wdev" down 2>/dev/null
-		iw dev "$wdev" del
+		iw dev "$wdev" del 2>/dev/null
 	done
 }
 
@@ -728,6 +737,9 @@ drv_mac80211_setup() {
 	json_get_values basic_rate_list basic_rate
 	json_select ..
 
+	# Restart the watchdog to avoid false positive triggers during reload
+	/etc/init.d/wifi-watchdog restart || true
+
 	find_phy || {
 		echo "Could not find PHY for device '$1'"
 		wireless_set_retry 0
@@ -782,15 +794,15 @@ drv_mac80211_setup() {
 	for_each_interface "ap" mac80211_prepare_vif
 
 	[ -n "$hostapd_ctrl" ] && {
-		/usr/sbin/hostapd -s -P /var/run/wifi-$phy.pid -B "$hostapd_conf_file"
-		ret="$?"
-		# Scott added to fix a problem with auto channel selection
-		sleep 5
-		wireless_add_process "$(cat /var/run/wifi-$phy.pid)" "/usr/sbin/hostapd" 1
-		[ "$ret" != 0 ] && {
-			wireless_setup_failed HOSTAPD_START_FAILED
-			return
-		}
+        	. $IPKG_INSTROOT/lib/functions/procd.sh >/dev/null 2>&1
+		procd_open_service hostapd-$phy
+		procd_open_instance $phy
+        	procd_set_param command /usr/sbin/hostapd \
+			-P /var/run/wifi-$phy.pid  \
+			"$hostapd_conf_file"
+        	procd_set_param respawn
+        	procd_close_instance
+		procd_close_service 2>/dev/null
 	}
 
 	for_each_interface "ap sta adhoc mesh monitor" mac80211_setup_vif
@@ -814,6 +826,8 @@ drv_mac80211_teardown() {
 	json_get_vars phy
 	json_select ..
 
+        . $IPKG_INSTROOT/lib/functions/procd.sh > /dev/null 2>&1
+	procd_kill hostapd-$phy $phy
 	mac80211_interface_cleanup "$phy"
 }
 
