--- a/feeds/routing/quagga/Makefile
+++ b/feeds/routing/quagga/Makefile
@@ -8,9 +8,9 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=quagga
-PKG_VERSION:=0.99.22.4
+PKG_VERSION:=0.99.23.1
 PKG_RELEASE:=1
-PKG_MD5SUM:=03ef24a448be47beba80efa2152f8a28
+PKG_MD5SUM:=da14aed6ae4be582486816f3eac2a46f
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
 PKG_SOURCE_URL:=http://download.savannah.gnu.org/releases/quagga/
