Index: build_dir/package/network/services/hostapd/files/hostapd.sh
===================================================================
--- build_dir.orig/package/network/services/hostapd/files/hostapd.sh
+++ build_dir/package/network/services/hostapd/files/hostapd.sh
@@ -507,14 +507,6 @@ append_hs20_conn_capab() {
 	[ -n "$1" ] && append bss_conf "hs20_conn_capab=$1" "$N"
 }
 
-append_radius_acct_req_attr() {
-	[ -n "$1" ] && append bss_conf "radius_acct_req_attr=$1" "$N"
-}
-
-append_radius_auth_req_attr() {
-	[ -n "$1" ] && append bss_conf "radius_auth_req_attr=$1" "$N"
-}
-
 append_airtime_sta_weight() {
 	[ -n "$1" ] && append bss_conf "airtime_sta_weight=$1" "$N"
 }
@@ -722,7 +714,16 @@ hostapd_set_bss_options() {
 				append bss_conf "radius_das_port=$dae_port" "$N"
 				append bss_conf "radius_das_client=$dae_client $dae_secret" "$N"
 			}
-			json_for_each_item append_radius_auth_req_attr radius_auth_req_attr
+
+			json_get_values radius_auth_req_attr radius_auth_req_attr
+			json_get_values radius_acct_req_attr radius_acct_req_attr
+
+			for attr in $radius_auth_req_attr; do
+				append bss_conf "radius_auth_req_attr=$attr" "$N"
+			done
+			for attr in $radius_acct_req_attr; do
+				append bss_conf "radius_acct_req_attr=$attr" "$N"
+			done
 
 			[ -n "$ownip" ] && append bss_conf "own_ip_addr=$ownip" "$N"
 			[ -n "$radius_client_addr" ] && append bss_conf "radius_client_addr=$radius_client_addr" "$N"
