--- a/feeds/packages/utils/collectd/files/collectd.init
+++ b/feeds/packages/utils/collectd/files/collectd.init
@@ -3,13 +3,14 @@
 
 START=80
 
-SERVICE_USE_PID=1
+USE_PROCD=1
+COLLECTD_BIN="/usr/sbin/collectd"
 
-start() {
-	mkdir -m 0755 -p /var/lib/collectd
-	service_start /usr/sbin/collectd
-}
-
-stop() {
-	service_stop /usr/sbin/collectd
+start_service()
+{
+        mkdir -m 0755 -p /var/lib/collectd
+        procd_open_instance
+        procd_set_param respawn
+        procd_set_param command "$COLLECTD_BIN" -f
+        procd_close_instance
 }
