--- a/feeds/packages/net/squid/Makefile
+++ b/feeds/packages/net/squid/Makefile
@@ -33,7 +33,7 @@ endef
 define Package/squid
   $(call Package/squid/Default)
   MENU:=1
-  DEPENDS:=+libopenssl +libpthread +librt +libltdl +libstdcpp
+  DEPENDS:=+libopenssl +libpthread +librt +libltdl +libstdcpp +libgnutls
   TITLE:=full-featured Web proxy cache
 endef
 
@@ -73,7 +73,7 @@ CONFIGURE_ARGS += \
 	--with-pthreads \
 	--without-expat \
 	--without-libxml2 \
-	--without-gnutls \
+	--with-gnutls \
 	--without-nettle \
 	--with-openssl=$(STAGING_DIR)/usr \
 	--enable-epoll \
