--- a/package/libs/openssl/Makefile
+++ b/package/libs/openssl/Makefile
@@ -14,11 +14,12 @@ PKG_USE_MIPS16:=0
 
 PKG_BUILD_PARALLEL:=0
 
+PKG_SOURCE_DIR:=old/1.0.2
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
-PKG_SOURCE_URL:=http://www.openssl.org/source/ \
-	ftp://ftp.openssl.org/source/ \
-	ftp://ftp.funet.fi/pub/crypt/mirrors/ftp.openssl.org/source \
-	ftp://ftp.sunet.se/pub/security/tools/net/openssl/source/
+PKG_SOURCE_URL:=http://www.openssl.org/source/$(PKG_SOURCE_DIR)/ \
+	ftp://ftp.openssl.org/source/$(PKG_SOURCE_DIR)/ \
+	ftp://ftp.funet.fi/pub/crypt/mirrors/ftp.openssl.org/source/$(PKG_SOURCE_DIR)/ \
+	ftp://ftp.sunet.se/pub/security/tools/net/openssl/source/$(PKG_SOURCE_DIR)/
 PKG_MD5SUM:=38dd619b2e77cbac69b99f52a053d25a
 
 PKG_LICENSE:=OpenSSL
