--- a/feeds/packages/net/openvswitch/Makefile
+++ b/feeds/packages/net/openvswitch/Makefile
@@ -12,25 +12,44 @@ include $(INCLUDE_DIR)/kernel.mk
 
 PKG_NAME:=openvswitch
 
-PKG_RELEASE:=2
+PKG_RELEASE:=1
 PKG_VERSION:=2.7.0
 PKG_RELEASE=$(PKG_SOURCE_VERSION)
 PKG_LICENSE:=Apache-2.0
 PKG_LICENSE_FILES:=COPYING
 PKG_USE_MIPS16:=0
 
-PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
-PKG_SOURCE_URL:=http://openvswitch.org/releases/
-PKG_HASH:=e492cf08a929b4a2178b7f9b01dc4ff562f44138b547b4e942078187b2445d2e
-
-PKG_BUILD_DEPENDS:=python/host python-six/host CONFIG_PACKAGE_openvswitch-python
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://github.com/openvswitch/ovs
+PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
+PKG_SOURCE_VERSION:=a03263a416fee837afebf0471a5e76b47d2161cc
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION)-$(PKG_SOURCE_VERSION).tar.gz
 
 PKG_BUILD_PARALLEL:=1
 PKG_FIXUP:=autoreconf
 PKG_INSTALL:=1
 
+# Upstream package supports kernels between 2.6.32 and 4.3 
+# see https://github.com/openvswitch/ovs/blob/master/FAQ.md
+# This list is pruned to only those kernels used in OpenWRT
+# Support for kernel 4.4 is backported
 SUPPORTED_KERNELS:=LINUX_3_18||LINUX_4_1||LINUX_4_3||LINUX_4_4||LINUX_4_9
 
+# If only kmod-openvswitch is enabled, then override default make path to only
+# build and install the datapath/linux subdirectory which cuts down compilation
+# time dramatically.
+ifeq ($(CONFIG_PACKAGE_openvswitch-base),)
+ ifneq ($(CONFIG_PACKAGE_kmod-openvswitch),)
+  MAKE_PATH := datapath/linux
+ endif
+endif
+
+# Additionally register PKG_CONFIG_DEPENDS to trigger a rebuild of the code
+# base if the package selection changes.
+PKG_CONFIG_DEPENDS := \
+	CONFIG_PACKAGE_openvswitch-base \
+	CONFIG_PACKAGE_kmod-openvswitch
+
 include $(INCLUDE_DIR)/package.mk
 $(call include_mk, python-package.mk)
 
@@ -67,13 +86,24 @@ endef
 define Package/openvswitch-python
   $(call Package/openvswitch/Default)
   TITLE:=Open vSwitch Python Support
-  DEPENDS:=+openvswitch +PACKAGE_openvswitch-python:python +PACKAGE_openvswitch-python:python-six
+  DEPENDS:=+openvswitch +PACKAGE_openvswitch-python:python
 endef
 
 define Package/openvswitch-python/description
   Provides bindings and libraries for using Python to manipulate/work with Open vSwitch.
 endef
 
+define Package/openvswitch-ipsec
+  $(call Package/openvswitch/Default)
+  TITLE:=Open vSwitch GRE through IPsec tool
+  DEPENDS:=+openvswitch-python
+endef
+
+define Package/openvswitch-ipsec/description
+  The ovs-monitor-ipsec script provides support for encrypting GRE tunnels with 
+  IPsec.
+endef
+
 OVS_BIN_TOOLS:=ovs-appctl ovs-ofctl ovs-dpctl ovs-vsctl ovsdb-client
 define Package/openvswitch
   $(call Package/openvswitch/Default)
@@ -91,16 +121,8 @@ define KernelPackage/openvswitch
   CATEGORY:=Kernel modules
   SUBMENU:=Network Support
   TITLE:=Open vSwitch Kernel Package
-  KCONFIG:= \
-	CONFIG_BRIDGE \
-	CONFIG_OPENVSWITCH \
-	CONFIG_OPENVSWITCH_GRE=y \
-	CONFIG_OPENVSWITCH_VXLAN=y \
-	CONFIG_OPENVSWITCH_GENEVE=y
-  DEPENDS:= \
-	@IPV6 +kmod-gre +kmod-lib-crc32c +kmod-mpls \
-	+kmod-vxlan +kmod-nf-nat +kmod-nf-nat6  \
-	@($(SUPPORTED_KERNELS))
+  KCONFIG:=CONFIG_BRIDGE
+  DEPENDS:=@IPV6 +kmod-gre +kmod-lib-crc32c +kmod-vxlan +kmod-nf-conntrack +kmod-nf-conntrack6 +kmod-nf-nat @($(SUPPORTED_KERNELS))
   FILES:= \
 	$(PKG_BUILD_DIR)/datapath/linux/openvswitch.$(LINUX_KMOD_SUFFIX) \
 	$(PKG_BUILD_DIR)/datapath/linux/vport-vxlan.$(LINUX_KMOD_SUFFIX) \
@@ -121,10 +143,7 @@ CONFIGURE_ARGS += --enable-shared
 TARGET_CFLAGS += -flto -std=gnu99
 
 CONFIGURE_VARS += KARCH=$(LINUX_KARCH)
-MAKE_FLAGS += \
-	ARCH="$(LINUX_KARCH)" \
-	PYTHONPATH="$(HOST_PYTHONPATH)" \
-	PYTHON="$(HOST_PYTHON_BIN)"
+MAKE_FLAGS += ARCH="$(LINUX_KARCH)"
 
 define OvsBinUtility
   define Package/openvswitch-$(1)
@@ -150,10 +169,10 @@ define Package/openvswitch-base/install
 	$(INSTALL_BIN) ./files/etc/init.d/openvswitch.init $(1)/etc/init.d/openvswitch
 
 	$(INSTALL_DIR) $(1)/usr/lib/
-	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libofproto*.so*  $(1)/usr/lib/
-	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libopenvswitch*.so*  $(1)/usr/lib/
-	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libovsdb*.so*  $(1)/usr/lib/
-	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libsflow*.so*  $(1)/usr/lib/
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libofproto-2.so*  $(1)/usr/lib/
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libopenvswitch-2.so*  $(1)/usr/lib/
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libovsdb-2.so*  $(1)/usr/lib/
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libsflow-2.so*  $(1)/usr/lib/
 
 	$(INSTALL_DIR) $(1)/usr/bin/
 	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/ovsdb-tool $(1)/usr/bin/
@@ -171,6 +190,11 @@ define Package/openvswitch-python/instal
 	$(CP) $(PKG_INSTALL_DIR)/usr/share/openvswitch/python/ovs/* $(1)/usr/lib/python$(PYTHON_VERSION)/ovs
 endef
 
+define Package/openvswitch-ipsec/install
+	$(INSTALL_DIR) $(1)/usr/sbin/
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/debian/ovs-monitor-ipsec $(1)/usr/sbin/
+endef
+
 define Package/openvswitch/install
 	:
 endef
@@ -185,6 +209,7 @@ $(foreach t,$(OVS_BIN_TOOLS),$(eval $(ca
 
 $(eval $(call BuildPackage,openvswitch-base))
 $(eval $(call BuildPackage,openvswitch-python))
+$(eval $(call BuildPackage,openvswitch-ipsec))
 $(eval $(call BuildPackage,openvswitch))
 $(eval $(call KernelPackage,openvswitch))
 
--- a/feeds/packages/net/openvswitch/patches/0003-override-pythonpath-via-make-vars.patch
+++ /dev/null
@@ -1,26 +0,0 @@
-diff --git a/Makefile.am b/Makefile.am
-index a4dd7b8..099a3c1 100644
---- a/Makefile.am
-+++ b/Makefile.am
-@@ -53,7 +53,7 @@ endif
- # foo/__init__.py into an (older) version with plain foo.py, since
- # foo/__init__.pyc will cause Python to ignore foo.py.
- run_python = \
--	PYTHONPATH=$(top_srcdir)/python$(psep)$$PYTHONPATH \
-+	PYTHONPATH=$(top_srcdir)/python$(psep)$(PYTHONPATH) \
- 	PYTHONDONTWRITEBYTECODE=yes $(PYTHON)
- 
- ALL_LOCAL =
-diff --git a/Makefile.in b/Makefile.in
-index 8df66e2..98b54bf 100644
---- a/Makefile.in
-+++ b/Makefile.in
-@@ -1654,7 +1654,7 @@ AM_CFLAGS = -Wstrict-prototypes $(WARNING_FLAGS) $(OVS_CFLAGS) \
- # foo/__init__.py into an (older) version with plain foo.py, since
- # foo/__init__.pyc will cause Python to ignore foo.py.
- run_python = \
--	PYTHONPATH=$(top_srcdir)/python$(psep)$$PYTHONPATH \
-+	PYTHONPATH=$(top_srcdir)/python$(psep)$(PYTHONPATH) \
- 	PYTHONDONTWRITEBYTECODE=yes $(PYTHON)
- 
- 
--- a/feeds/packages/net/openvswitch/patches/0005-disable-module-build.patch
+++ /dev/null
@@ -1,25 +0,0 @@
-From 7cf69704374e363fdda227803be3849233ba96f3 Mon Sep 17 00:00:00 2001
-From: Dario Ernst <Dario.Ernst@riverbed.com>
-Date: Fri, 25 Nov 2016 15:33:48 +0100
-Subject: [PATCH 4/4] disable-module-build
-
----
- datapath/linux/Makefile.main.in | 2 +-
- 1 file changed, 1 insertion(+), 1 deletion(-)
-
-diff --git a/datapath/linux/Makefile.main.in b/datapath/linux/Makefile.main.in
-index 7d18253..e5555a7 100644
---- a/datapath/linux/Makefile.main.in
-+++ b/datapath/linux/Makefile.main.in
-@@ -68,7 +68,7 @@ ifeq (,$(wildcard $(CONFIG_FILE)))
- endif
- 
- default:
--	$(MAKE) -C $(KSRC) M=$(builddir) modules
-+	@echo skipping module build
- 
- modules_install:
- 	$(MAKE) -C $(KSRC) M=$(builddir) modules_install
--- 
-2.9.3
-
--- /dev/null
+++ b/feeds/packages/net/openvswitch/patches/0008-add-back-old-gfp-this-node-define.patch
@@ -0,0 +1,19 @@
+diff --git a/datapath/flow.c b/datapath/flow.c
+index a7a2063..8db8041 100644
+--- a/datapath/flow.c
++++ b/datapath/flow.c
+@@ -51,6 +51,14 @@
+ 
+ #include "vlan.h"
+ 
++#ifndef GFP_THISNODE
++#ifdef CONFIG_NUMA  
++#define GFP_THISNODE    (__GFP_THISNODE | __GFP_NOWARN | __GFP_NORETRY)
++#else                             
++#define GFP_THISNODE    ((__force gfp_t)0)
++#endif 
++#endif
++
+ u64 ovs_flow_used_time(unsigned long flow_jiffies)
+ {
+ 	struct timespec cur_ts;
