Index: build_dir/package/kernel/ath10k-ct/Makefile
===================================================================
--- build_dir.orig/package/kernel/ath10k-ct/Makefile
+++ build_dir/package/kernel/ath10k-ct/Makefile
@@ -1,21 +1,21 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=ath10k-ct
-PKG_RELEASE=3
+PKG_RELEASE=1
 
 PKG_LICENSE:=GPLv2
 PKG_LICENSE_FILES:=
 
 PKG_SOURCE_URL:=https://github.com/greearb/ath10k-ct.git
 PKG_SOURCE_PROTO:=git
-PKG_SOURCE_DATE:=2018-09-29
-PKG_SOURCE_VERSION:=b9989fbd5d6e3269b3e49ab3ec71b4174a02209e
+PKG_SOURCE_DATE:=2018-11-02
+PKG_SOURCE_VERSION:=6c3a6262317f3240013014e79a570693009c0235
 PKG_MIRROR_HASH:=67bb1c81518aea5880757d521023db25190d9cca21ba67bf1c7e577efd287430
 
 # Build the 4.13 ath10k-ct driver version.  Other options are "-4.9", or
 # leave un-defined for 4.7 kernel.  Probably this should match as closely as
 # possible to whatever mac80211 backports version is being used.
-CT_KVER="-4.16"
+CT_KVER="-4.19"
 
 PKG_MAINTAINER:=Ben Greear <greearb@candelatech.com>
 PKG_BUILD_PARALLEL:=1
@@ -69,6 +69,7 @@ ifdef CONFIG_PACKAGE_MAC80211_MESH
   NOSTDINC_FLAGS += -DCONFIG_MAC80211_MESH
 endif
 
+CT_MAKEDEFS += CONFIG_ATH10K_CE=m
 CT_MAKEDEFS += CONFIG_ATH10K=m CONFIG_ATH10K_PCI=m
 
 # This AHB logic is needed for IPQ4019 radios
Index: build_dir/package/kernel/ath10k-ct/patches/1013-scott-ath10k-ct-remove-debug-wmi-command-0-not-supported.patch
===================================================================
--- /dev/null
+++ build_dir/package/kernel/ath10k-ct/patches/1013-scott-ath10k-ct-remove-debug-wmi-command-0-not-supported.patch
@@ -0,0 +1,12 @@
+Index: ath10k-ct-2018-11-02-6c3a6262/ath10k-4.19/wmi.c
+===================================================================
+--- ath10k-ct-2018-11-02-6c3a6262.orig/ath10k-4.19/wmi.c
++++ ath10k-ct-2018-11-02-6c3a6262/ath10k-4.19/wmi.c
+@@ -370,7 +370,6 @@ static struct wmi_cmd_map wmi_10x_cmd_ma
+ 	.set_cca_params_cmdid = WMI_CMD_UNSUPPORTED,
+ 	.pdev_bss_chan_info_request_cmdid =
+ 		WMI_10_2_PDEV_BSS_CHAN_INFO_REQUEST_CMDID,
+-	.pdev_bss_chan_info_request_cmdid = WMI_CMD_UNSUPPORTED,
+ 	.pdev_get_tpc_table_cmdid = WMI_CMD_UNSUPPORTED,
+ 	.radar_found_cmdid = WMI_CMD_UNSUPPORTED,
+ };
