Index: build_dir/package/kernel/ath10k-ct/Makefile
===================================================================
--- build_dir.orig/package/kernel/ath10k-ct/Makefile
+++ build_dir/package/kernel/ath10k-ct/Makefile
@@ -15,7 +15,7 @@ PKG_MIRROR_HASH:=dc1097f3a7b4b7e346918f2
 # Build the 5.2 ath10k-ct driver version.  Other option is "-4.19".
 # Probably this should match as closely as
 # possible to whatever mac80211 backports version is being used.
-CT_KVER="-4.19"
+CT_KVER="-5.2"
 
 PKG_MAINTAINER:=Ben Greear <greearb@candelatech.com>
 PKG_BUILD_PARALLEL:=1
