--- a/feeds/packages/net/mwan3/files/etc/hotplug.d/iface/15-mwan3
+++ b/feeds/packages/net/mwan3/files/etc/hotplug.d/iface/15-mwan3
@@ -15,6 +15,9 @@ mwan3_set_connected_iptables
 
 config_load mwan3
 config_get enabled $INTERFACE enabled 0
+# Check dynamic interfaces
+[ "$enabled" == "1" ] || INTERFACE="${INTERFACE%_4}" config_get enabled $INTERFACE enabled 0
+[ "$enabled" == "1" ] || INTERFACE="${INTERFACE%_6}" config_get enabled $INTERFACE enabled 0
 [ "$enabled" == "1" ] || exit 0
 
 if [ "$ACTION" == "ifup" ]; then
