Index: build_dir/feeds/routing/batctl/Makefile
===================================================================
--- build_dir.orig/feeds/routing/batctl/Makefile
+++ build_dir/feeds/routing/batctl/Makefile
@@ -42,12 +42,12 @@ define Package/batctl-default
 $(call Package/batctl/Default)
   TITLE:=B.A.T.M.A.N. Advanced user space configuration tool (Default)
   VARIANT:=default
-  ALTERNATIVES:=200:/usr/sbin/batctl:/usr/libexec/batctl-full
+  ALTERNATIVES:=200:/usr/sbin/batctl:/usr/libexec/batctl-default
 endef
 
 define Package/batctl-full
 $(call Package/batctl/Default)
-  TITLE:=B.A.T.M.A.N. Advanced user space configuration tool (Minimal)
+  TITLE:=B.A.T.M.A.N. Advanced user space configuration tool (Full)
   VARIANT:=full
   ALTERNATIVES:=300:/usr/sbin/batctl:/usr/libexec/batctl-full
 endef
