Index: build_dir/feeds/packages/net/coova-chilli/files/chilli.init
===================================================================
--- build_dir.orig/feeds/packages/net/coova-chilli/files/chilli.init
+++ build_dir/feeds/packages/net/coova-chilli/files/chilli.init
@@ -9,7 +9,7 @@ service_triggers() {
 }
 
 config_cb() {
-	local chilli_inst="$2"
+	chilli_inst="$2"
 	if [ "$chilli_inst" != "" ]; then
 		chilli_conf="/var/run/chilli_${chilli_inst}.conf"
 		if [ -e "$chilli_conf" ]; then
@@ -29,7 +29,7 @@ option_cb() {
 			echo "dhcpif=\"$ifname\"" >> "$chilli_conf"
 			;;
 		disabled)
-			[ "$(config_get_bool "$1")" = "1" ] && eval "start_chilli_$chilli_inst=0"
+			[ "$2" = "1" ] && eval "start_chilli_$chilli_inst=0"
 			;;
 		# boolean settings
 		debug|dhcpbroadcast|nodynip|vlanlocation|locationstopstart|locationcopycalled|locationimmediateupdate|locationopt82|coanoipcheck|noradallow|proxymacaccept|proxyonacct|dhcpmacset|dhcpradius|noc2c|eapolenable|uamanydns|uamanyip|uamnatanyip|nouamsuccess|nowispr1|nowispr2|domaindnslocal|radsec|macauth|macreauth|macauthdeny|macallowlocal|strictmacauth|strictdhcp|ieee8021q|only8021q|radiusoriginalurl|swapoctets|statusfilesave|wpaguests|openidauth|papalwaysok|mschapv2|chillixml|acctupdate|dnsparanoia|seskeepalive|usetap|noarpentries|framedservice|scalewin|redir|injectwispr|redirurl|routeonetone|nousergardendata|uamgardendata|uamotherdata|withunixipc|uamallowpost|redirssl|uamuissl|layer3|patricia|redirdnsreq|dhcpnotidle|ipv6|ipv6only)
