Index: build_dir/package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh
===================================================================
--- build_dir.orig/package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh
+++ build_dir/package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh
@@ -782,6 +782,7 @@ drv_mac80211_setup() {
 		rxantenna txantenna \
 		frag rts beacon_int:100 htmode
 	json_get_values basic_rate_list basic_rate
+	json_get_vars ifname
 	json_select ..
 
 	find_phy || {
@@ -853,6 +854,13 @@ drv_mac80211_setup() {
         	procd_set_param respawn
         	procd_close_instance
 		procd_close_service 2>/dev/null
+		procd_open_service watchdog-$phy
+		procd_open_instance $phy
+        	procd_set_param command /sbin/wifi-watchdog \
+			$ifname $phy
+        	procd_set_param respawn
+        	procd_close_instance
+		procd_close_service 2>/dev/null
 	}
 
 	for_each_interface "ap sta adhoc mesh monitor" mac80211_setup_vif
@@ -879,6 +887,7 @@ drv_mac80211_teardown() {
         . $IPKG_INSTROOT/lib/functions/procd.sh > /dev/null 2>&1
 	procd_kill hostapd-$phy $phy
 	procd_kill wpa_supplicant-$phy $phy
+	procd_kill watchdog-$phy $phy
 	mac80211_interface_cleanup "$phy"
 }
 
