Index: build_dir/feeds/packages/net/mwan3/files/usr/sbin/mwan3
===================================================================
--- build_dir.orig/feeds/packages/net/mwan3/files/usr/sbin/mwan3
+++ build_dir/feeds/packages/net/mwan3/files/usr/sbin/mwan3
@@ -108,6 +108,10 @@ status()
 }
 
 start() {
+	config_get_bool enabled globals 'enabled' 0
+	[ ${enabled} -gt 0 ] || {
+		exit 1
+	}
 	/etc/init.d/mwan3 enable
 	/etc/init.d/mwan3 start
 }
