Index: build_dir/feeds/packages/net/bonding/files/lib/netifd/proto/bonding.sh
===================================================================
--- build_dir.orig/feeds/packages/net/bonding/files/lib/netifd/proto/bonding.sh
+++ build_dir/feeds/packages/net/bonding/files/lib/netifd/proto/bonding.sh
@@ -79,19 +79,25 @@ proto_bonding_add_slave() {
 		return
 	fi
 
-	ip link set dev "$slave" down
+	let retries=5
+	while [ $((retries)) -gt 0 ] ; do
+		ip link set dev "$slave" down
+		echo "+$slave" > /sys/class/net/"$link"/bonding/slaves && break
+		let retries=retries-1
+		sleep 1
+	done
 
-	sleep 1
-
-	echo "+$slave" > /sys/class/net/"$link"/bonding/slaves
-
-	ip link set dev "$slave" up
+	ifup "$slave"
 }
 
 proto_bonding_setup() {
 	local cfg="$1"
 	local link="bond-$cfg"
 
+	json_get_vars ifname
+	[ -z "$ifname" ] || link="$ifname"
+
+
 	# Check for loaded kernel bonding driver (/sys/class/net/bonding_masters exists)
 	[ -f "$BONDING_MASTERS" ] || {
 		echo "$cfg" "setup: bonding_masters does not exist in sysfs (kernel module not loaded?)"
@@ -193,16 +199,10 @@ proto_bonding_setup() {
 
 	proto_init_update "$link" 1
 
-	# For static configuration we _MUST_ have an IP address
-	[ -z "$ipaddr" ] && {
-		echo "$cfg" "No local IP address defined"
-		proto_notify_error "$cfg" INVALID_LOCAL_ADDRESS
-		proto_block_restart "$cfg"
-		return
+	[ -z "$ipaddr" ] || {
+		proto_add_ipv4_address "$ipaddr" "$netmask"
 	}
 
-	proto_add_ipv4_address "$ipaddr" "$netmask"
-
 	proto_send_update "$cfg"
 }
 
@@ -210,6 +210,9 @@ proto_bonding_teardown() {
 	local cfg="$1"
 	local link="bond-$cfg"
 
+	json_get_vars ifname
+	[ -z "$ifname" ] || link="$ifname"
+
 	# Check for loaded kernel bonding driver (/sys/class/net/bonding_masters exists)
 	[ -f "$BONDING_MASTERS" ] || {
 		echo "$cfg" "teardown: bonding_masters does not exist in sysfs (kernel module not loaded?)"
