Index: build_dir/feeds/packages/net/frr/files/daemons
===================================================================
--- build_dir.orig/feeds/packages/net/frr/files/daemons
+++ build_dir/feeds/packages/net/frr/files/daemons
@@ -1,9 +1,9 @@
 # The watchfrr and zebra daemons are always started.
 #
-bgpd=no
-ospfd=no
+bgpd=yes
+ospfd=yes
 #ospfd_instances=1,20
-ospf6d=no
+ospf6d=yes
 ripd=no
 ripngd=no
 isisd=no
@@ -14,9 +14,9 @@ eigrpd=no
 babeld=no
 sharpd=no
 pbrd=no
-bfdd=no
+bfdd=yes
 fabricd=no
-vrrpd=no
+vrrpd=yes
 
 #
 # If this option is set the /etc/init.d/frr script automatically loads
Index: build_dir/feeds/packages/net/frr/files/frr
===================================================================
--- build_dir.orig/feeds/packages/net/frr/files/frr
+++ build_dir/feeds/packages/net/frr/files/frr
@@ -19,8 +19,20 @@ else
 	. "/usr/sbin/frrcommon.sh"
 fi
 
+migrate_quagga_configs() {
+	if [ -d /etc/quagga ]; then
+		logger "Migrating Quagga configuration files to FRR..."
+		for file in /etc/quagga/*.conf ; do
+			logger "Moving file $file"
+			mv "$file" /etc/frr/
+		done
+		logger "Remaining Quagga files can be found in /etc/quagga.bak"
+		mv /etc/quagga /etc/quagga.bak
+	fi
+}
 
 start() {
+        migrate_quagga_configs
 	daemon_list daemons
 	watchfrr_options="$watchfrr_options $daemons"
 	daemon_start watchfrr
Index: build_dir/feeds/packages/net/frr/Makefile
===================================================================
--- build_dir.orig/feeds/packages/net/frr/Makefile
+++ build_dir/feeds/packages/net/frr/Makefile
@@ -207,7 +207,7 @@ define Package/frr-libfrr/install
 	$(INSTALL_DIR) $(1)/usr/lib $(1)/etc/frr
 	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libfrr.so* $(1)/usr/lib/
 	$(if $(CONFIG_FRR_SNMP),$(CP) $(PKG_INSTALL_DIR)/usr/lib/libfrrsnmp.so* $(1)/usr/lib/,)
-	$(INSTALL_CONF) ./files/{frr.conf,daemons} $(1)/etc/frr/
+	$(INSTALL_CONF) ./files/daemons $(1)/etc/frr/
 endef
 
 define Package/frr-vtysh/install
Index: build_dir/feeds/packages/net/frr/files/frr.conf
===================================================================
--- build_dir.orig/feeds/packages/net/frr/files/frr.conf
+++ /dev/null
@@ -1,22 +0,0 @@
-password zebra
-!
-!router eigrp 1
-! network 10.0.0.0/8
-! network 192.168.1.0/24
-!
-!router ospf
-!ospf router-id 172.16.0.2
-!network 192.168.1.0/24 area 0
-!neighbor 172.16.0.1
-!
-!router rip
-! network 10.0.0.0/8
-! network 192.168.1.0/24
-!
-log syslog
-!
-access-list vty permit 127.0.0.0/8
-access-list vty deny any
-!
-line vty
- access-class vty
Index: build_dir/feeds/packages/net/frr/files/frrcommon.sh
===================================================================
--- build_dir.orig/feeds/packages/net/frr/files/frrcommon.sh
+++ build_dir/feeds/packages/net/frr/files/frrcommon.sh
@@ -282,6 +282,8 @@ load_old_config() {
 	for dmn in $DAEMONS; do eval "test \$_new_$dmn != no && $dmn=\$_new_$dmn; unset _new_$dmn"; done
 }
 
+chownfrr "$C_PATH"
+
 [ -r "$C_PATH/daemons" ] || {
 	log_failure_msg "cannot run $@: $C_PATH/daemons does not exist"
 	exit 1
