Index: build_dir/package/firmware/linux-firmware/broadcom.mk
===================================================================
--- build_dir.orig/package/firmware/linux-firmware/broadcom.mk
+++ build_dir/package/firmware/linux-firmware/broadcom.mk
@@ -79,6 +79,15 @@ define Package/brcmfmac-firmware-43455-s
 endef
 $(eval $(call BuildPackage,brcmfmac-firmware-43455-sdio-rpi-4b))
 
+Package/brcmfmac-firmware-43455-sdio-rpi-cm4 = $(call Package/firmware-default,Broadcom BCM43455 NVRAM for Raspberry Pi CM4)
+define Package/brcmfmac-firmware-43455-sdio-rpi-cm4/install
+	$(INSTALL_DIR) $(1)/lib/firmware/brcm
+	$(INSTALL_DATA) \
+		$(PKG_BUILD_DIR)/brcm/brcmfmac43455-sdio.raspberrypi,4-model-b.txt \
+		$(1)/lib/firmware/brcm/brcmfmac43455-sdio.raspberrypi,4-compute-module.txt
+endef
+$(eval $(call BuildPackage,brcmfmac-firmware-43455-sdio-rpi-cm4))
+
 Package/brcmfmac-firmware-usb = $(call Package/firmware-default,Broadcom BCM43xx fullmac USB firmware)
 define Package/brcmfmac-firmware-usb/install
 	$(INSTALL_DIR) $(1)/lib/firmware/brcm
Index: build_dir/target/linux/bcm27xx/image/Makefile
===================================================================
--- build_dir.orig/target/linux/bcm27xx/image/Makefile
+++ build_dir/target/linux/bcm27xx/image/Makefile
@@ -183,7 +183,7 @@ define Device/apb3500
 	raspberrypi,4-model-b
   DEVICE_PACKAGES := \
 	cypress-firmware-43455-sdio \
-	cypress-nvram-43455-sdio-rpi-4b \
+	brcmfmac-firmware-43455-sdio-rpi-cm4 \
 	kmod-brcmfmac wpad-basic-wolfssl
   IMAGE/sysupgrade.img.gz := boot-common | boot-2711 | sdcard-img | gzip | append-metadata
   IMAGE/factory.img.gz := boot-common | boot-2711 | sdcard-img | gzip
@@ -203,7 +203,7 @@ define Device/apb3350
 	raspberrypi,4-model-b
   DEVICE_PACKAGES := \
 	cypress-firmware-43455-sdio \
-	cypress-nvram-43455-sdio-rpi-4b \
+	brcmfmac-firmware-43455-sdio-rpi-cm4 \
 	kmod-brcmfmac wpad-basic-wolfssl
   IMAGE/sysupgrade.img.gz := boot-common | boot-2711 | sdcard-img | gzip | append-metadata
   IMAGE/factory.img.gz := boot-common | boot-2711 | sdcard-img | gzip
