Index: build_dir.ap2100/package/firmware/linux-firmware/Makefile
===================================================================
--- build_dir.ap2100.orig/package/firmware/linux-firmware/Makefile
+++ build_dir.ap2100/package/firmware/linux-firmware/Makefile
@@ -11,8 +11,8 @@ PKG_NAME:=linux-firmware
 PKG_RELEASE:=1
 
 PKG_SOURCE_PROTO:=git
-PKG_SOURCE_DATE:=2018-12-16
-PKG_SOURCE_VERSION:=211de1679a68b8ab0f841a8058df35e13e3963f0
+PKG_SOURCE_DATE:=2019-02-21
+PKG_SOURCE_VERSION:=54b0a748c8966c93aaa8726402e0b69cb51cd5d2
 PKG_MIRROR_HASH:=894a13a44c9de58528488682feb3ade582725cd766a16ccaca07089628229d90
 PKG_SOURCE_URL:=https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git
 
