--- a/ip/iptunnel.c
+++ b/ip/iptunnel.c
@@ -38,6 +38,7 @@ static void usage(void)
 	fprintf(stderr, "          [ prl-default ADDR ] [ prl-nodefault ADDR ] [ prl-delete ADDR ]\n");
 	fprintf(stderr, "          [ 6rd-prefix ADDR ] [ 6rd-relay_prefix ADDR ] [ 6rd-reset ]\n");
 	fprintf(stderr, "          [ ttl TTL ] [ tos TOS ] [ [no]pmtudisc ] [ dev PHYS_DEV ]\n");
+        fprintf(stderr, "          [ keepalive NUMBER ] [ retries NUMBER ]\n");
 	fprintf(stderr, "\n");
 	fprintf(stderr, "Where: NAME := STRING\n");
 	fprintf(stderr, "       ADDR := { IP_ADDRESS | any }\n");
@@ -203,6 +204,18 @@ static int parse_args(int argc, char **a
 					invarg("bad TOS value", *argv);
 				p->iph.tos |= uval;
 			}
+                } else if (strcmp(*argv, "keepalive") == 0) {
+                        unsigned uval;
+                        NEXT_ARG();
+                        if (get_unsigned(&uval, *argv, 0))
+                                invarg("invalid keepalive value\n", *argv);
+                        p->ka_interval = uval;
+                } else if (strcmp(*argv, "retries") == 0) {
+                        unsigned uval;
+                        NEXT_ARG();
+                        if (get_unsigned(&uval, *argv, 0))
+                                invarg("invalid retries value\n", *argv);
+                        p->ka_retries = uval;
 		} else {
 			if (strcmp(*argv, "name") == 0) {
 				NEXT_ARG();
--- a/include/linux/if_tunnel.h
+++ b/include/linux/if_tunnel.h
@@ -35,6 +35,10 @@ struct ip_tunnel_parm {
 	__be32			i_key;
 	__be32			o_key;
 	struct iphdr		iph;
+        struct timer_list               *timer;
+        int             ka_interval;
+        int             ka_retries;
+        int             ka_cur_retries;
 };
 
 enum {
