--- GobiNet/Structs.h.orig	2012-11-16 09:10:42.354390435 -0600
+++ GobiNet/Structs.h	2012-11-16 09:11:12.330391584 -0600
@@ -44,6 +44,7 @@
 //---------------------------------------------------------------------------
 // Include Files
 //---------------------------------------------------------------------------
+#include <linux/module.h>
 #include <linux/etherdevice.h>
 #include <linux/ethtool.h>
 #include <linux/mii.h>
--- GobiSerial/GobiSerial.c.orig	2012-11-16 09:11:19.914391874 -0600
+++ GobiSerial/GobiSerial.c	2012-11-16 09:11:49.686393014 -0600
@@ -44,6 +44,7 @@
 // Include Files
 //---------------------------------------------------------------------------
 
+#include <linux/module.h>
 #include <linux/tty.h>
 #include <linux/tty_flip.h>
 #include <linux/usb.h> 
