Index: build_dir/tools/firmware-utils/src/mkmylofw.c
===================================================================
--- build_dir.orig/tools/firmware-utils/src/mkmylofw.c
+++ build_dir/tools/firmware-utils/src/mkmylofw.c
@@ -158,6 +158,10 @@ struct cpx_board boards[] = {
 		"WPE72", "Compex WPE72",
 		"BareBoard"),
 
+	CPX_BOARD_AR71XX(DEVID_COMPEX_WPJ344, 16,
+		"WPJ344", "Compex WPJ344",
+		"BareBoard"),
+
 	CPX_BOARD_AR23XX(DEVID_COMPEX_NP25G, 4,
 		"NP25G", "Compex NetPassage 25G",
 		"Wireless 54Mbps XR Router"),
Index: build_dir/tools/firmware-utils/src/myloader.h
===================================================================
--- build_dir.orig/tools/firmware-utils/src/myloader.h
+++ build_dir/tools/firmware-utils/src/myloader.h
@@ -63,6 +63,7 @@
 /* Devices based on the Atheros AR71xx */
 #define DEVID_COMPEX_WP543	0x0640
 #define DEVID_COMPEX_WPE72	0x0672
+#define DEVID_COMPEX_WPJ344	0x0687
 
 /* Devices based on the IXP422 */
 #define DEVID_COMPEX_WP18	0x047E
Index: build_dir/target/linux/ar71xx/base-files/etc/board.d/02_network
===================================================================
--- build_dir.orig/target/linux/ar71xx/base-files/etc/board.d/02_network
+++ build_dir/target/linux/ar71xx/base-files/etc/board.d/02_network
@@ -561,9 +561,13 @@ ar71xx_setup_interfaces()
 		ucidef_add_switch_port_attr "switch0" 5 led 2
 		;;
 	wpj344)
-		ucidef_add_switch "switch0" \
-			"0@eth0" "3:lan" "2:wan"
-		;;
+        	ucidef_set_interfaces_lan_wan "eth0.4001" "eth0.4000"
+        	ucidef_add_switch "switch0" \
+                	"3:lan:1:4001" "2:wan:2:4000" "0@eth0"
+        	ucidef_add_switch_extra_attr "switch0" "alias_cpu" "0"
+        	ucidef_add_switch_extra_attr "switch0" "alias_wan" "2"
+        	ucidef_add_switch_extra_attr "switch0" "alias_lan" "3"
+        	;;
 	wpj558)
 		ucidef_add_switch "switch0" \
 			"5:lan" "1:wan" "6@eth0"
Index: build_dir/target/linux/ar71xx/image/legacy.mk
===================================================================
--- build_dir.orig/target/linux/ar71xx/image/legacy.mk
+++ build_dir/target/linux/ar71xx/image/legacy.mk
@@ -643,7 +643,6 @@ Image/Build/CyberTANGZIP=$(call Image/Bu
 Image/Build/CyberTANLZMA/buildkernel=$(call MkuImageLzma,$(2),$(3) $(4))
 Image/Build/CyberTANLZMA=$(call Image/Build/CyberTAN,$(1),$(2),$(3),$(4),$(5))
 
-
 Image/Build/Netgear/initramfs=$(call MkuImageLzma/initramfs,$(2),$(3) $(4),,-M $(5))
 
 define Image/Build/Netgear/buildkernel
Index: build_dir/target/linux/ar71xx/image/generic.mk
===================================================================
--- build_dir.orig/target/linux/ar71xx/image/generic.mk
+++ build_dir/target/linux/ar71xx/image/generic.mk
@@ -1257,9 +1257,32 @@ endef
 TARGET_DEVICES += wlr8100
 
 define Device/wpj-16m
+  $(Device/CpxImageParams16M)
   DEVICE_PACKAGES := kmod-usb2 kmod-usb-ledtrig-usbport
   MTDPARTS := spi0.0:192k(u-boot)ro,16128k(firmware),64k(art)ro
   IMAGE_SIZE := 16128k
+  IMAGES := sysupgrade.bin factory.img
+  IMAGE/sysupgrade.bin = append-kernel | pad-to $$$$(BLOCKSIZE) | append-rootfs | pad-rootfs | check-size $$$$(IMAGE_SIZE)
+  IMAGE/factory.img = Cpxfactory16M
+endef
+
+define Device/CpxImageParams16M
+  KSTART := 30000
+  RSTART := 1b0000
+  P1START := fe0000
+  P2START := ff0000
+  FIRMWARE_SIZE := 0x1000000
+  KERNEL_SIZE := 1572864
+  ROOTFS_SIZE := 14876672
+endef
+
+define Build/Cpxfactory16M
+        $(STAGING_DIR_HOST)/bin/mkmylofw -B $(BOARDNAME) -s $(FIRMWARE_SIZE) \
+                -p0x$(KSTART):$(KERNEL_SIZE):al:0x80060000:$(BOARDNAME):$(IMAGE_KERNEL) \
+                -p0x$(RSTART):$(ROOTFS_SIZE):::rootfs:$(IMAGE_ROOTFS) \
+                -p0x$(P1START):0x010000 \
+                -p0x$(P2START):0x010000 \
+                $@
 endef
 
 define Device/wpj342
