--- a/target/linux/ar71xx/base-files/etc/uci-defaults/02_network
+++ b/target/linux/ar71xx/base-files/etc/uci-defaults/02_network
@@ -70,7 +70,6 @@ cpe510)
 	ucidef_add_switch_vlan "switch0" "2" "0t 4"
 	;;
 
-db120 |\
 rb-2011l | \
 rb-2011uas |\
 rb-2011uias |\
@@ -444,6 +443,15 @@ wndr3700)
 	uci set network.@switch_port[-1].led='2'
 	;;
 
+wpj344)
+	ucidef_set_interfaces_lan_wan "eth0.4001" "eth0.4000"
+	ucidef_add_switch "switch0" "1" "1"
+	ucidef_add_switch_vlan "switch0" "0" "0t 2"
+	uci set network.@switch_vlan[-1].vid='4000'
+	ucidef_add_switch_vlan "switch0" "1" "0t 3"
+	uci set network.@switch_vlan[-1].vid='4001'
+	;;
+
 *)
 	ucidef_set_interfaces_lan_wan "eth0" "eth1"
 	;;
