--- a/target/linux/ar71xx/base-files/etc/board.d/02_network
+++ b/target/linux/ar71xx/base-files/etc/board.d/02_network
@@ -437,8 +437,14 @@ wpj531)
 	;;
 
 wpj558)
-	ucidef_add_switch "switch0" \
-		"5:lan" "1:wan" "6@eth0"
+        ucidef_set_interfaces_lan_wan "eth0.4001" "eth0.4000"
+        ucidef_add_switch "switch0" "1" "1"
+	uci set network.@switch[-1].alias_cpu='6'
+	uci set network.@switch[-1].alias_wan='5'
+	uci set network.@switch[-1].alias_lan='1'
+        ucidef_set_interfaces_vlans_lan_wan "4001" "4000"
+        ucidef_add_switch_vlan "switch0" "4000" "5 6t"
+        ucidef_add_switch_vlan "switch0" "4001" "1 6t"
 	;;
 
 ap121 |\
--- a/tools/firmware-utils/src/mkmylofw.c
+++ b/tools/firmware-utils/src/mkmylofw.c
@@ -161,6 +161,9 @@ struct cpx_board boards[] = {
 	CPX_BOARD_AR71XX(DEVID_COMPEX_WPJ344, 16,
 		"WPJ344", "Compex WPJ344",
 		"BareBoard"),
+	CPX_BOARD_AR71XX(DEVID_COMPEX_WPJ558, 16,
+		"WPJ558", "Compex WPJ558",
+		"BareBoard"),
 
 	CPX_BOARD_AR23XX(DEVID_COMPEX_NP25G, 4,
 		"NP25G", "Compex NetPassage 25G",
--- a/tools/firmware-utils/src/myloader.h
+++ b/tools/firmware-utils/src/myloader.h
@@ -64,6 +64,7 @@
 #define DEVID_COMPEX_WP543	0x0640
 #define DEVID_COMPEX_WPE72	0x0672
 #define DEVID_COMPEX_WPJ344	0x0687
+#define DEVID_COMPEX_WPJ558	0x0691
 
 /* Devices based on the IXP422 */
 #define DEVID_COMPEX_WP18	0x047E
--- a/target/linux/ar71xx/image/legacy.mk
+++ b/target/linux/ar71xx/image/legacy.mk
@@ -1032,7 +1032,7 @@ $(eval $(call SingleProfile,AthLzma,64k,
 $(eval $(call SingleProfile,Cpxwrt,64k,WPJ344_16M,wpj344-16M,WPJ344,ttyS0,115200,$$(wpj344_mtdlayout_16M),KRuImage,65536,0x1000000,1572864,14876672,WPJ344))
 $(eval $(call SingleProfile,AthLzma,64k,DR344,dr344,DR344,ttyS0,115200,$$(dr344_mtdlayout),RKuImage))
 $(eval $(call SingleProfile,AthLzma,64k,WPJ531_16M,wpj531-16M,WPJ531,ttyS0,115200,$$(wpj531_mtdlayout_16M),KRuImage,65536))
-$(eval $(call SingleProfile,AthLzma,64k,WPJ558_16M,wpj558-16M,WPJ558,ttyS0,115200,$$(wpj558_mtdlayout_16M),KRuImage,65536))
+$(eval $(call SingleProfile,Cpxwrt,64k,WPJ558_16M,wpj558-16M,WPJ558,ttyS0,115200,$$(wpj558_mtdlayout_16M),KRuImage,65536,0x1000000,1572864,14876672,WPJ558))
 $(eval $(call SingleProfile,AthLzma,64k,WRTNODE2Q,wrtnode2q,WRTNODE2Q,ttyS0,115200,$$(wrtnode2q_mtdlayout),KRuImage))
 $(eval $(call SingleProfile,AthLzma,64k,YUN_8M,yun-8M,Yun,ttyATH0,250000,$$(yun_mtdlayout_8M),RKuImage))
 $(eval $(call SingleProfile,AthLzma,64k,YUN_16M,yun-16M,Yun,ttyATH0,250000,$$(yun_mtdlayout_16M),RKuImage))
