--- a/target/linux/ar71xx/base-files/etc/board.d/02_network
+++ b/target/linux/ar71xx/base-files/etc/board.d/02_network
@@ -468,9 +468,13 @@ ar71xx_setup_interfaces()
         	ucidef_add_switch_extra_attr "switch0" "alias_lan" "3"
         	;;
 	wpj558)
-		ucidef_add_switch "switch0" \
-			"5:lan" "1:wan" "6@eth0"
-		;;
+	        ucidef_set_interfaces_lan_wan "eth0.4001" "eth0.4000"
+        	ucidef_add_switch "switch0" \
+                	"1:lan:1:4001" "5:wan:2:4000" "6@eth0"
+        	ucidef_add_switch_extra_attr "switch0" "alias_cpu" "6"
+        	ucidef_add_switch_extra_attr "switch0" "alias_wan" "5"
+        	ucidef_add_switch_extra_attr "switch0" "alias_lan" "1"
+        	;;
 	wpj563)
 		ucidef_add_switch "switch0" \
 			"0@eth0" "2:lan" "3:wan"
--- a/tools/firmware-utils/src/mkmylofw.c
+++ b/tools/firmware-utils/src/mkmylofw.c
@@ -161,6 +161,9 @@ struct cpx_board boards[] = {
 	CPX_BOARD_AR71XX(DEVID_COMPEX_WPJ344, 16,
 		"WPJ344", "Compex WPJ344",
 		"BareBoard"),
+	CPX_BOARD_AR71XX(DEVID_COMPEX_WPJ558, 16,
+		"WPJ558", "Compex WPJ558",
+		"BareBoard"),
 
 	CPX_BOARD_AR23XX(DEVID_COMPEX_NP25G, 4,
 		"NP25G", "Compex NetPassage 25G",
--- a/tools/firmware-utils/src/myloader.h
+++ b/tools/firmware-utils/src/myloader.h
@@ -64,6 +64,7 @@
 #define DEVID_COMPEX_WP543	0x0640
 #define DEVID_COMPEX_WPE72	0x0672
 #define DEVID_COMPEX_WPJ344	0x0687
+#define DEVID_COMPEX_WPJ558	0x0691
 
 /* Devices based on the IXP422 */
 #define DEVID_COMPEX_WP18	0x047E
--- a/target/linux/ar71xx/image/generic.mk
+++ b/target/linux/ar71xx/image/generic.mk
@@ -841,11 +841,15 @@ endef
 TARGET_DEVICES += wpj531
 
 define Device/wpj558
+  $(Device/CpxImageParams16M)
   DEVICE_TITLE := Compex WPJ558 (16MB flash)
   DEVICE_PACKAGES := kmod-usb-core kmod-usb-ohci kmod-usb2 kmod-usb-ledtrig-usbport
   BOARDNAME := WPJ558
   MTDPARTS := spi0.0:192k(u-boot)ro,16128k(firmware),64k(art)ro
   IMAGE_SIZE := 16128k
+  IMAGES := sysupgrade.bin factory.img
+  IMAGE/sysupgrade.bin = append-kernel | pad-to $$$$(BLOCKSIZE) | append-rootfs | pad-rootfs | check-size $$$$(IMAGE_SIZE)
+  IMAGE/factory.img = Cpxfactory16M
 endef
 TARGET_DEVICES += wpj558
 
