Index: build_dir/tools/firmware-utils/src/mkmylofw.c
===================================================================
--- build_dir.orig/tools/firmware-utils/src/mkmylofw.c
+++ build_dir/tools/firmware-utils/src/mkmylofw.c
@@ -161,6 +161,9 @@ struct cpx_board boards[] = {
 	CPX_BOARD_AR71XX(DEVID_COMPEX_WPJ344, 16,
 		"WPJ344", "Compex WPJ344",
 		"BareBoard"),
+	CPX_BOARD_AR71XX(DEVID_COMPEX_WPJ558, 16,
+		"WPJ558", "Compex WPJ558",
+		"BareBoard"),
 
 	CPX_BOARD_AR23XX(DEVID_COMPEX_NP25G, 4,
 		"NP25G", "Compex NetPassage 25G",
Index: build_dir/tools/firmware-utils/src/myloader.h
===================================================================
--- build_dir.orig/tools/firmware-utils/src/myloader.h
+++ build_dir/tools/firmware-utils/src/myloader.h
@@ -64,6 +64,7 @@
 #define DEVID_COMPEX_WP543	0x0640
 #define DEVID_COMPEX_WPE72	0x0672
 #define DEVID_COMPEX_WPJ344	0x0687
+#define DEVID_COMPEX_WPJ558	0x0691
 
 /* Devices based on the IXP422 */
 #define DEVID_COMPEX_WP18	0x047E
