--- a/target/linux/ar71xx/image/Makefile
+++ b/target/linux/ar71xx/image/Makefile
@@ -2301,6 +2301,7 @@ $(eval $(call SingleProfile,AthLzma,64k,
 $(eval $(call SingleProfile,AthLzma,64k,WLR8100,wlr8100,WLR8100,ttyS0,115200,$$(wlr8100_mtdlayout),KRuImage))
 $(eval $(call SingleProfile,Cpxwrt,64k,WPJ344_16M,wpj344-16M,WPJ344,ttyS0,115200,$$(wpj344_mtdlayout_16M),KRuImage,65536,0x1000000,1572864,14876672,WPJ344))
 $(eval $(call SingleProfile,Cpxwrt,64k,WPJ558_16M,wpj558-16M,WPJ558,ttyS0,115200,$$(wpj558_mtdlayout_16M),KRuImage,65536,0x1000000,1572864,14876672,WPJ558))
+$(eval $(call SingleProfile,Cpxwrt,64k,WPJ531_16M,wpj531-16M,WPJ531,ttyS0,115200,$$(wpj531_mtdlayout_16M),KRuImage,65536,0x1000000,1572864,14876672,WPJ531))
 $(eval $(call SingleProfile,AthLzma,64k,YUN_8M,yun-8M,Yun,ttyATH0,250000,$$(yun_mtdlayout_8M),RKuImage))
 $(eval $(call SingleProfile,AthLzma,64k,YUN_16M,yun-16M,Yun,ttyATH0,250000,$$(yun_mtdlayout_16M),RKuImage))
 
--- a/tools/firmware-utils/src/mkmylofw.c
+++ b/tools/firmware-utils/src/mkmylofw.c
@@ -164,6 +164,9 @@ struct cpx_board boards[] = {
 	CPX_BOARD_AR71XX(DEVID_COMPEX_WPJ558, 16,
 		"WPJ558", "Compex WPJ558",
 		"BareBoard"),
+	CPX_BOARD_AR71XX(DEVID_COMPEX_WPJ531, 16,
+		"WPJ531", "Compex WPJ531",
+		"BareBoard"),
 
 	CPX_BOARD_AR23XX(DEVID_COMPEX_NP25G, 4,
 		"NP25G", "Compex NetPassage 25G",
--- a/tools/firmware-utils/src/myloader.h
+++ b/tools/firmware-utils/src/myloader.h
@@ -65,6 +65,7 @@
 #define DEVID_COMPEX_WPE72	0x0672
 #define DEVID_COMPEX_WPJ344	0x0687
 #define DEVID_COMPEX_WPJ558	0x0691
+#define DEVID_COMPEX_WPJ531	0x0693
 
 /* Devices based on the IXP422 */
 #define DEVID_COMPEX_WP18	0x047E
--- a/target/linux/ar71xx/base-files/etc/board.d/02_network
+++ b/target/linux/ar71xx/base-files/etc/board.d/02_network
@@ -438,7 +438,7 @@ wpj344)
 	;;
 
 wpj531)
-	ucidef_set_interfaces_lan_wan "eth0" "eth1"
+	ucidef_set_interfaces_lan_wan "eth1" "eth0"
 	;;
 
 wpj558)
