--- a/package/kernel/mac80211/Makefile
+++ b/package/kernel/mac80211/Makefile
@@ -549,7 +549,7 @@ define KernelPackage/ath5k
   URL:=http://linuxwireless.org/en/users/Drivers/ath5k
   DEPENDS+= @PCI_SUPPORT||@TARGET_atheros +kmod-ath
   FILES:=$(PKG_BUILD_DIR)/drivers/net/wireless/ath/ath5k/ath5k.ko
-  AUTOLOAD:=$(call AutoProbe,ath5k)
+  AUTOLOAD:=$(call AutoLoad,52,ath5k)
 endef
 
 define KernelPackage/ath5k/description
@@ -574,7 +574,7 @@ define KernelPackage/ath9k
   DEPENDS+= @PCI_SUPPORT||TARGET_ar71xx +kmod-ath9k-common
   FILES:= \
 	$(PKG_BUILD_DIR)/drivers/net/wireless/ath/ath9k/ath9k.ko
-  AUTOLOAD:=$(call AutoProbe,ath9k)
+  AUTOLOAD:=$(call AutoLoad,53,ath9k)
 endef
 
 define KernelPackage/ath9k/description
@@ -597,7 +597,7 @@ define KernelPackage/ath9k-htc
   DEPENDS+= @USB_SUPPORT +kmod-ath9k-common +kmod-usb-core
   FILES:= \
 	$(PKG_BUILD_DIR)/drivers/net/wireless/ath/ath9k/ath9k_htc.ko
-  AUTOLOAD:=$(call AutoProbe,ath9k_htc)
+  AUTOLOAD:=$(call AutoLoad,54,ath9k_htc)
 endef
 
 define KernelPackage/ath9k-htc/description
