--- a/include/host-build.mk
+++ b/include/host-build.mk
@@ -131,7 +131,7 @@ define Host/Exports/Default
   $(1) : export STAGING_PREFIX=$$(STAGING_DIR_HOST)
   $(1) : export PKG_CONFIG_PATH=$$(STAGING_DIR_HOST)/lib/pkgconfig
   $(1) : export PKG_CONFIG_LIBDIR=$$(STAGING_DIR_HOST)/lib/pkgconfig
-  $(1) : export CCACHE_DIR:=$(STAGING_DIR_HOST)/ccache
+  $(1) : export CCACHE_DIR?=$(STAGING_DIR_HOST)/ccache
 endef
 Host/Exports=$(Host/Exports/Default)
 
--- a/include/package.mk
+++ b/include/package.mk
@@ -147,7 +147,7 @@ define Build/Exports/Default
   $(1) : export CONFIG_SITE:=$$(CONFIG_SITE)
   $(1) : export PKG_CONFIG_PATH:=$$(PKG_CONFIG_PATH)
   $(1) : export PKG_CONFIG_LIBDIR:=$$(PKG_CONFIG_PATH)
-  $(1) : export CCACHE_DIR:=$(STAGING_DIR)/ccache
+  $(1) : export CCACHE_DIR?=$(STAGING_DIR)/ccache
 endef
 Build/Exports=$(Build/Exports/Default)
 
