Index: build_dir/target/linux/generic/config-4.14
===================================================================
--- build_dir.orig/target/linux/generic/config-4.14
+++ build_dir/target/linux/generic/config-4.14
@@ -3105,7 +3105,7 @@ CONFIG_NET_SCHED=y
 # CONFIG_NET_SCH_CBQ is not set
 # CONFIG_NET_SCH_CHOKE is not set
 # CONFIG_NET_SCH_CODEL is not set
-# CONFIG_NET_SCH_DEFAULT is not set
+CONFIG_NET_SCH_DEFAULT=y
 # CONFIG_NET_SCH_DRR is not set
 # CONFIG_NET_SCH_DSMARK is not set
 CONFIG_NET_SCH_FIFO=y
Index: build_dir/target/linux/generic/config-4.19
===================================================================
--- build_dir.orig/target/linux/generic/config-4.19
+++ build_dir/target/linux/generic/config-4.19
@@ -3266,7 +3266,7 @@ CONFIG_NET_SCHED=y
 # CONFIG_NET_SCH_CBS is not set
 # CONFIG_NET_SCH_CHOKE is not set
 # CONFIG_NET_SCH_CODEL is not set
-# CONFIG_NET_SCH_DEFAULT is not set
+CONFIG_NET_SCH_DEFAULT=y
 # CONFIG_NET_SCH_DRR is not set
 # CONFIG_NET_SCH_DSMARK is not set
 # CONFIG_NET_SCH_ETF is not set
