Index: build_dir/feeds/routing/nodogsplash/Makefile
===================================================================
--- build_dir.orig/feeds/routing/nodogsplash/Makefile
+++ build_dir/feeds/routing/nodogsplash/Makefile
@@ -7,12 +7,12 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=nodogsplash
 PKG_VERSION:=5.0.0
-PKG_RELEASE:=1
+PKG_RELEASE:=2
 
 PKG_SOURCE_URL:=https://github.com/imagestream/nodogsplash.git
 PKG_SOURCE_PROTO:=git
-PKG_SOURCE_DATE:=2020-02-04
-PKG_SOURCE_VERSION:=8519b8d44664cb45ffefd0fb6470e58babac2b52
+PKG_SOURCE_DATE:=2022-04-08
+PKG_SOURCE_VERSION:=b8268253d203b2780329a7b5cd712691f4302798
 
 PKG_MAINTAINER:=Moritz Warning <moritzwarning@web.de>
 PKG_LICENSE:=GPL-2.0-or-later
