--- a/configure.ac	2018-04-24 12:05:46.122679906 +0300
+++ b/configure.ac	2018-04-24 18:50:18.885120833 +0300
@@ -514,7 +514,7 @@
   AC_CHECK_TOOLS([PYTHONCONFIG], [python3-config python-config python2-config])
   if test -n "$PYTHONCONFIG"; then
     PYTHON_CFLAGS="`\"${PYTHONCONFIG}\" --includes`"
-    PYTHON_LIBS="`\"${PYTHONCONFIG}\" --libs`"
+    PYTHON_LIBS="`\"${PYTHONCONFIG}\" --libs` -lz"
 
     AC_MSG_CHECKING([whether we found a working Python version])
     AC_LINK_IFELSE_FLAGS([$PYTHON_CFLAGS], [$PYTHON_LIBS], [AC_LANG_PROGRAM([
