--- a/package/network/utils/iproute2/Makefile
+++ b/package/network/utils/iproute2/Makefile
@@ -91,7 +91,7 @@ $(call Package/iproute2/Default)
 endef
 
 ifeq ($(BUILD_VARIANT),tiny)
-  IP_CONFIG_TINY:=y
+  IP_CONFIG_TINY:=n
 endif
 
 ifdef CONFIG_PACKAGE_devlink
@@ -115,7 +115,7 @@ endef
 TARGET_CFLAGS += -ffunction-sections -fdata-sections
 
 MAKE_FLAGS += \
-	EXTRA_CCOPTS="-I$(STAGING_DIR)/usr/include/libnl-tiny" \
+	EXTRA_CCOPTS="-I$(STAGING_DIR)/usr/include/libnl" \
 	KERNEL_INCLUDE="$(LINUX_DIR)/include" \
 	SHARED_LIBS="" \
 	LDFLAGS="$(TARGET_LDFLAGS) -Wl,--gc-sections" \
