Index: build_dir/package/kernel/ath10k-ct/patches/1002-scott-ath10k-set-mac-address-from-wpj344-hack.patch
===================================================================
--- /dev/null
+++ build_dir/package/kernel/ath10k-ct/patches/1002-scott-ath10k-set-mac-address-from-wpj344-hack.patch
@@ -0,0 +1,25 @@
+--- a/ath10k-4.19/mac.c
++++ b/ath10k-4.19/mac.c
+@@ -9007,6 +9007,10 @@ static const struct ieee80211_tpt_blink
+ };
+ #endif
+ 
++#ifdef CONFIG_ATH10K_MAC_OVERRIDE
++extern u8 *ath10k_macaddr;
++#endif
++
+ int ath10k_mac_register(struct ath10k *ar)
+ {
+ 	static const u32 cipher_suites[] = {
+@@ -9020,6 +9025,11 @@ int ath10k_mac_register(struct ath10k *a
+ 	void *channels;
+ 	int ret;
+ 
++#ifdef CONFIG_ATH10K_MAC_OVERRIDE
++        if (ath10k_macaddr)
++                memcpy(ar->mac_addr, ath10k_macaddr, ETH_ALEN);
++#endif
++
+ 	SET_IEEE80211_PERM_ADDR(ar->hw, ar->mac_addr);
+ 
+ 	SET_IEEE80211_DEV(ar->hw, ar->dev);
Index: build_dir/package/kernel/ath10k-ct/Makefile
===================================================================
--- build_dir.orig/package/kernel/ath10k-ct/Makefile
+++ build_dir/package/kernel/ath10k-ct/Makefile
@@ -45,6 +45,17 @@ define KernelPackage/ath10k-ct/config
                depends on PACKAGE_kmod-ath10k-ct
 endef
 
+define KernelPackage/ath10k-ct/config
+  if PACKAGE_kmod-ath10k-ct
+       config ATH10K_MAC_OVERRIDE
+		bool "Override ath10k MAC address from board"
+                default n
+                help
+                  Override the ath10k card's MAC address with one pre-allocated by the motherboard
+  endif
+endef
+
+
 NOSTDINC_FLAGS = \
 	-I$(PKG_BUILD_DIR) \
 	-I$(STAGING_DIR)/usr/include/mac80211-backport/uapi \
@@ -90,6 +101,10 @@ ifeq ($(CONFIG_ATH10K-CT_LEDS),y)
   NOSTDINC_FLAGS += -DCONFIG_ATH10K_LEDS
 endif
 
+ifdef CONFIG_ATH10K_MAC_OVERRIDE
+  NOSTDINC_FLAGS += -DCONFIG_ATH10K_MAC_OVERRIDE
+endif
+
 define Build/Configure
 	cp $(STAGING_DIR)/usr/include/mac80211/ath/*.h $(PKG_BUILD_DIR)
 endef
