--- a/feeds/packages/net/openvswitch/Makefile
+++ b/feeds/packages/net/openvswitch/Makefile
@@ -94,15 +94,18 @@ define KernelPackage/openvswitch
   KCONFIG:= \
 	CONFIG_BRIDGE \
 	CONFIG_OPENVSWITCH \
-	CONFIG_OPENVSWITCH_GRE=n \
-	CONFIG_OPENVSWITCH_VXLAN=n \
-	CONFIG_OPENVSWITCH_GENEVE=n
+	CONFIG_OPENVSWITCH_GRE=y \
+	CONFIG_OPENVSWITCH_VXLAN=y \
+	CONFIG_OPENVSWITCH_GENEVE=y
   DEPENDS:= \
 	@IPV6 +kmod-gre +kmod-lib-crc32c +kmod-mpls \
 	+kmod-vxlan +kmod-nf-nat +kmod-nf-nat6  \
 	@($(SUPPORTED_KERNELS))
-  FILES:= $(LINUX_DIR)/net/openvswitch/openvswitch.ko
-  AUTOLOAD:=$(call AutoLoad,21,openvswitch)
+  FILES:= \
+	$(PKG_BUILD_DIR)/datapath/linux/openvswitch.$(LINUX_KMOD_SUFFIX) \
+	$(PKG_BUILD_DIR)/datapath/linux/vport-vxlan.$(LINUX_KMOD_SUFFIX) \
+	$(PKG_BUILD_DIR)/datapath/linux/vport-gre.$(LINUX_KMOD_SUFFIX)
+  AUTOLOAD:=$(call AutoLoad,21,openvswitch vport-vxlan vport-gre)
 endef
 
 define KernelPackage/openvswitch/description
