Index: build_dir/target/linux/generic/config-4.14
===================================================================
--- build_dir.orig/target/linux/generic/config-4.14
+++ build_dir/target/linux/generic/config-4.14
@@ -3105,12 +3105,18 @@ CONFIG_NET_SCHED=y
 # CONFIG_NET_SCH_CBQ is not set
 # CONFIG_NET_SCH_CHOKE is not set
 # CONFIG_NET_SCH_CODEL is not set
-# CONFIG_NET_SCH_DEFAULT is not set
+CONFIG_NET_SCH_DEFAULT=y
 # CONFIG_NET_SCH_DRR is not set
 # CONFIG_NET_SCH_DSMARK is not set
 CONFIG_NET_SCH_FIFO=y
 # CONFIG_NET_SCH_FQ is not set
 CONFIG_NET_SCH_FQ_CODEL=y
+CONFIG_DEFAULT_FQ_CODEL=y
+CONFIG_DEFAULT_NET_SCH="fq_codel"
+# CONFIG_DEFAULT_FQ is not set
+# CONFIG_DEFAULT_CODEL is not set
+# CONFIG_DEFAULT_SFQ is not set
+# CONFIG_DEFAULT_PFIFO_FAST is not set
 # CONFIG_NET_SCH_GRED is not set
 # CONFIG_NET_SCH_HFSC is not set
 # CONFIG_NET_SCH_HHF is not set
Index: build_dir/target/linux/generic/config-4.19
===================================================================
--- build_dir.orig/target/linux/generic/config-4.19
+++ build_dir/target/linux/generic/config-4.19
@@ -3266,13 +3266,19 @@ CONFIG_NET_SCHED=y
 # CONFIG_NET_SCH_CBS is not set
 # CONFIG_NET_SCH_CHOKE is not set
 # CONFIG_NET_SCH_CODEL is not set
-# CONFIG_NET_SCH_DEFAULT is not set
+CONFIG_NET_SCH_DEFAULT=y
 # CONFIG_NET_SCH_DRR is not set
 # CONFIG_NET_SCH_DSMARK is not set
 # CONFIG_NET_SCH_ETF is not set
 CONFIG_NET_SCH_FIFO=y
 # CONFIG_NET_SCH_FQ is not set
 CONFIG_NET_SCH_FQ_CODEL=y
+CONFIG_DEFAULT_FQ_CODEL=y
+CONFIG_DEFAULT_NET_SCH="fq_codel"
+# CONFIG_DEFAULT_FQ is not set
+# CONFIG_DEFAULT_CODEL is not set
+# CONFIG_DEFAULT_SFQ is not set
+# CONFIG_DEFAULT_PFIFO_FAST is not set
 # CONFIG_NET_SCH_GRED is not set
 # CONFIG_NET_SCH_HFSC is not set
 # CONFIG_NET_SCH_HHF is not set
