--- a/net/bridge/br_input.c
+++ b/net/bridge/br_input.c
@@ -168,7 +168,7 @@ rx_handler_result_t br_handle_frame(stru
 	if (unlikely(skb->pkt_type == PACKET_LOOPBACK))
 		return RX_HANDLER_PASS;
 
-	if (!is_valid_ether_addr(eth_hdr(skb)->h_source))
+	if (!is_valid_ether_addr(eth_hdr(skb)->h_source) && !is_zero_ether_addr(eth_hdr(skb)->h_source))
 		goto drop;
 
 	skb = skb_share_check(skb, GFP_ATOMIC);
