--- a/drivers/usb/class/cdc-wdm.c
+++ b/drivers/usb/class/cdc-wdm.c
@@ -579,6 +579,7 @@ static int wdm_open(struct inode *inode,
 	struct usb_interface *intf;
 	struct wdm_device *desc;
 
+retry:
 	mutex_lock(&wdm_mutex);
 	desc = wdm_find_device_by_minor(minor);
 	if (!desc)
@@ -597,6 +598,15 @@ static int wdm_open(struct inode *inode,
 
 	/* using write lock to protect desc->count */
 	mutex_lock(&desc->wlock);
+	if (desc->count != 0) {
+//		printk("Waiting on device count %d (pid %d)\n", desc->count, task_pid_nr(current));
+		usb_autopm_put_interface(desc->intf);
+		mutex_unlock(&desc->wlock);
+		mutex_unlock(&wdm_mutex);
+		if ((schedule_timeout_interruptible(HZ)) > 0)
+			return -EBUSY;
+		goto retry;
+	}
 	if (!desc->count++) {
 		desc->werr = 0;
 		desc->rerr = 0;
