Index: build_dir/target/linux/generic/config-4.14
===================================================================
--- build_dir.orig/target/linux/generic/config-4.14
+++ build_dir/target/linux/generic/config-4.14
@@ -2752,6 +2752,7 @@ CONFIG_MODULE_UNLOAD=y
 # CONFIG_MPL115_SPI is not set
 # CONFIG_MPL3115 is not set
 # CONFIG_MPLS is not set
+# CONFIG_MPLS_ROUTING is not set
 # CONFIG_MPU3050_I2C is not set
 # CONFIG_MQ_IOSCHED_DEADLINE is not set
 # CONFIG_MQ_IOSCHED_KYBER is not set
Index: build_dir/target/linux/generic/config-4.19
===================================================================
--- build_dir.orig/target/linux/generic/config-4.19
+++ build_dir/target/linux/generic/config-4.19
@@ -2901,6 +2901,7 @@ CONFIG_MODULE_UNLOAD=y
 # CONFIG_MPL115_SPI is not set
 # CONFIG_MPL3115 is not set
 # CONFIG_MPLS is not set
+# CONFIG_MPLS_ROUTING is not set
 # CONFIG_MPU3050_I2C is not set
 # CONFIG_MQ_IOSCHED_DEADLINE is not set
 # CONFIG_MQ_IOSCHED_KYBER is not set
