--- a/modules/admin-full/luasrc/model/cbi/admin_network/vlan.lua
+++ b/modules/admin-full/luasrc/model/cbi/admin_network/vlan.lua
@@ -130,8 +130,8 @@ m.uci:foreach("network", "switch",
 			for _, section in luci.util.spairs(
 				osections,
 				function(a, b)
-					return (tonumber(m:get(osections[a], has_vlan4k or "vlan")) or 9999)
-						<  (tonumber(m:get(osections[b], has_vlan4k or "vlan")) or 9999)
+					return (tonumber(m:get(osections[a], "vlan")) or 9999)
+						<  (tonumber(m:get(osections[b], "vlan")) or 9999)
 				end
 			) do
 				sections[#sections+1] = section
@@ -202,7 +202,7 @@ m.uci:foreach("network", "switch",
 		end
 
 
-		local vid = s:option(Value, has_vlan4k or "vlan", "VLAN ID", "<div id='portstatus-%s'></div>" % switch_name)
+		local vid = s:option(Value, "vlan", "VLAN ID", "<div id='portstatus-%s'></div>" % switch_name)
 		local mx_vid = has_vlan4k and 4094 or (num_vlans - 1)
 
 		vid.rmempty = false
@@ -254,7 +254,7 @@ m.uci:foreach("network", "switch",
 
 		-- Fallback to "vlan" option if "vid" option is supported but unset.
 		vid.cfgvalue = function(self, section)
-			return m:get(section, has_vlan4k or "vlan")
+			return m:get(section, "vlan")
 				or m:get(section, "vlan")
 		end
 
