--- a/modules/admin-full/luasrc/model/cbi/admin_network/wifi.lua
+++ b/modules/admin-full/luasrc/model/cbi/admin_network/wifi.lua
@@ -386,7 +386,9 @@ mode = s:taboption("general", ListValue,
 mode.override_values = true
 mode:value("ap", translate("Access Point"))
 mode:value("sta", translate("Client"))
-mode:value("adhoc", translate("Ad-Hoc"))
+if not iw.hwmodelist.ac then
+	mode:value("adhoc", translate("Ad-Hoc"))
+end
 
 bssid = s:taboption("general", Value, "bssid", translate("<abbr title=\"Basic Service Set Identifier\">BSSID</abbr>"))
 
@@ -450,11 +452,13 @@ end
 -------------------- MAC80211 Interface ----------------------
 
 if hwtype == "mac80211" then
-	if fs.access("/usr/sbin/iw") then
+	if fs.access("/usr/sbin/iw") and not iw.hwmodelist.ac then
 		mode:value("mesh", "802.11s Mesh")
 	end
 
-	mode:value("ahdemo", translate("Pseudo Ad-Hoc (ahdemo)"))
+	if not iw.hwmodelist.ac then
+		mode:value("ahdemo", translate("Pseudo Ad-Hoc (ahdemo)"))
+	end
 	mode:value("monitor", translate("Monitor"))
 	bssid:depends({mode="adhoc"})
 	bssid:depends({mode="sta"})
