Index: luci/applications/luci-app-statistics/root/etc/config/luci_statistics
===================================================================
--- luci.orig/applications/luci-app-statistics/root/etc/config/luci_statistics
+++ luci/applications/luci-app-statistics/root/etc/config/luci_statistics
@@ -40,7 +40,7 @@ config statistics 'collectd_logfile'
 	option Timestamp '1'
 
 config statistics 'collectd_network'
-	option enable '0'
+	option enable '1'
 
 config statistics 'collectd_unixsock'
 	option enable '0'
@@ -55,24 +55,24 @@ config statistics 'collectd_apcups'
 	option Port '3551'
 
 config statistics 'collectd_conntrack'
-	option enable '0'
+	option enable '1'
 
 config statistics 'collectd_contextswitch'
 	option enable '0'
 
 config statistics 'collectd_cpu'
 	option enable '1'
+	option ValuesPercentage '1'
 
 config statistics 'collectd_cpufreq'
-	option enable '0'
+	option enable '1'
 
 config statistics 'collectd_curl'
 	option enable '0'
 
 config statistics 'collectd_df'
-	option enable '0'
-	option Devices '/dev/mtdblock/4'
-	option MountPoints '/jffs'
+	option enable '1'
+	option MountPoints '/overlay'
 	option FSTypes 'tmpfs'
 	option IgnoreSelected '0'
 
@@ -83,7 +83,6 @@ config statistics 'collectd_disk'
 
 config statistics 'collectd_dns'
 	option enable '0'
-	option Interfaces 'br-lan'
 	option IgnoreSources '127.0.0.1'
 
 config statistics 'collectd_entropy'
@@ -94,7 +93,6 @@ config statistics 'collectd_exec'
 
 config statistics 'collectd_interface'
 	option enable '1'
-	option Interfaces 'br-lan'
 	option IgnoreSelected '0'
 
 config statistics 'collectd_iptables'
@@ -132,10 +130,8 @@ config statistics 'collectd_memory'
 	option ValuesPercentage '0'
 
 config statistics 'collectd_netlink'
-	option enable '0'
+	option enable '1'
 	option IgnoreSelected '0'
-	option VerboseInterfaces 'br-lan'
-	option QDiscs 'br-lan'
 
 config statistics 'collectd_nut'
 	option enable '0'
@@ -147,14 +143,13 @@ config statistics 'collectd_olsrd'
 	option Host '127.0.0.1'
 
 config statistics 'collectd_ping'
-	option enable '0'
+	option enable '1'
 	option TTL '127'
 	option Interval '30'
-	option Hosts '127.0.0.1'
+	option Hosts '8.8.8.8 8.8.4.4'
 
 config statistics 'collectd_processes'
-	option enable '0'
-	option Processes 'uhttpd dnsmasq dropbear'
+	option enable '1'
 
 config statistics 'collectd_sensors'
 	option enable '0'
@@ -163,9 +158,9 @@ config statistics 'collectd_splash_lease
 	option enable '0'
 
 config statistics 'collectd_tcpconns'
-	option enable '0'
+	option enable '1'
 	option ListeningPorts '0'
-	option LocalPorts '22 80'
+	option LocalPorts '22 80 443'
 
 config statistics 'collectd_thermal'
 	option enable '0'
@@ -173,5 +168,5 @@ config statistics 'collectd_thermal'
 	option Device ''
 
 config statistics 'collectd_uptime'
-	option enable '0'
+	option enable '1'
 
Index: luci/applications/luci-app-statistics/luasrc/model/cbi/luci_statistics/cpu.lua
===================================================================
--- luci.orig/applications/luci-app-statistics/luasrc/model/cbi/luci_statistics/cpu.lua
+++ luci/applications/luci-app-statistics/luasrc/model/cbi/luci_statistics/cpu.lua
@@ -33,4 +33,8 @@ valuespercentage = s:option( Flag, "Valu
 valuespercentage.default = 0
 valuespercentage:depends({ enable = 1, ReportByCpu = 1, ReportByState = 1 })
 
+valuespercentage = s:option( Flag, "ValuesPercentage", translate("Use percentages instead of jiffies") )
+valuespercentage.default = 1
+valuespercentage:depends( "enable", 1 )
+
 return m
