Index: luci/modules/luci-mod-system/luasrc/controller/admin/system.lua
===================================================================
--- luci.orig/modules/luci-mod-system/luasrc/controller/admin/system.lua
+++ luci/modules/luci-mod-system/luasrc/controller/admin/system.lua
@@ -243,7 +243,8 @@ function action_restore()
 	if upload and #upload > 0 then
 		if os.execute("gunzip -t %q >/dev/null 2>&1" % archive_tmp) == 0 then
 			luci.template.render("admin_system/applyreboot")
-			os.execute("tar -C / -xzf %q >/dev/null 2>&1" % archive_tmp)
+			os.execute("tar -C / -xzf %q >/dev/null 2>&1 || tar -C / -xf %q > /dev/null 2>&1" %{archive_tmp,archive_tmp})
+
 			luci.sys.reboot()
 		else
 			luci.template.render("admin_system/flashops", {
