Index: luci/protocols/luci-proto-wireguard/luasrc/model/cbi/admin_network/proto_wireguard.lua
===================================================================
--- luci.orig/protocols/luci-proto-wireguard/luasrc/model/cbi/admin_network/proto_wireguard.lua
+++ luci/protocols/luci-proto-wireguard/luasrc/model/cbi/admin_network/proto_wireguard.lua
@@ -2,6 +2,7 @@
 -- Licensed to the public under the Apache License 2.0.
 
 
+local sys = require "luci.sys"
 local map, section, net = ...
 local ifname = net:get_interface():name()
 local private_key, listen_port
@@ -21,6 +22,7 @@ private_key = section:taboption(
 private_key.password = true
 private_key.datatype = "and(base64,rangelength(44,44))"
 private_key.optional = false
+private_key.default = sys.exec("wg genkey")
 
 
 listen_port = section:taboption(
